include/h2o.h

```diff
@@ -628,6 +628,10 @@ typedef struct st_h2o_header_t {
      * value of the header
      */
     h2o_iovec_t value;
+    /**
+     * The name of the header as originally received from the client
+     */
+    const char *orig_hname;
```

Comment:
Maybe the order of the fields should be `name` -> `orig_hname` -> `value`?

I think we should consider `orig_hname` as a first-class member of `h2o_header_t` (we need to _always_ take care of the value!). In this respect, having fields for the name first makes more sense to me.

Also, it might be better to rename the field to `orig_name`. We do not need _h_ since this struct represents a header.